File: quivi.guests.react.app/src/hooks/api/Dtos/merchants/Fee.ts
import type { ChargeMethod } from "../ChargeMethod";
import type { FeeType } from "./FeeType";
import type { FeeUnit } from "./FeeUnit";

export interface Fee {
    readonly feeType: FeeType;
    readonly chargeMethod: ChargeMethod;
    readonly value: number;
    readonly unit: FeeUnit;
}

File: Quivi.Guests.Api/Dtos/Fee.cs
using Quivi.Domain.Entities.Charges;
using Quivi.Domain.Entities.Merchants;

namespace Quivi.Guests.Api.Dtos
{
    public class Fee
    {
        public FeeType Type { get; init; }
        public ChargeMethod PaymentMethod { get; init; }
        public decimal Value { get; init; }
        public FeeUnit Unit { get; init; }
    }
}